### PR TITLE
chore: remove legacy swapper 🩸

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Removes all code related to the old swapper.

This is not strictly a refactor, as some logic crossed domains or had to be re-written.

Also deletes a number of other unused code paths where it made sense to do so.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Medium. Whilst this is mostly a refactor, some logic needede to be (re-)written where concerns were not propertly separated. I've reimplemented the same logic, but human error is a risk.

## Testing

The swapper will need a complete regression test, and the app in general will need a once-over to ensure nothing unrelated was accidently touched.

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5015** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5015" target="_blank">Graphite</a> 
    * **PR #5016** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5016" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5016?utm_source=stack-comment).

File: package.json [hunk 1]
```diff
@@ -191,8 +191,7 @@
     "uuid": "^9.0.0",
     "viem": "^0.3.3",
     "web-vitals": "^2.1.4",
-    "web3": "1.7.4",
-    "zustand": "^4.3.6"
```
Comment (woodenfurniture) [inline package.json:195]: 🪦

File: src/state/apis/swappers/helpers/getInputOutputRatioFromQuote.ts [hunk 2]
```diff
@@ -128,6 +128,18 @@ const _convertCryptoBaseUnitToUsdPrecision = (
...
+  by the total fiat value of the sell amount (including network fees).
+
+  Higher ratios are better.
+
+  E.g. if the fiat value of the sell amount is 100 and the fiat value of the receive amount is 90,
+  the ratio is 0.9.
+
+  Negative ratios are possible when the fees exceed the sell amount.
+  This is allowed to let us choose 'the best from a bad bunch'.
+*/
```
Comment (woodenfurniture) [inline src/state/apis/swappers/helpers/getInputOutputRatioFromQuote.ts:142]: ❤️

File: src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts [hunk 3]
```diff
@@ -87,26 +74,7 @@ export const useTradeExecution = ({
...
-          return { ...oneInchSwapper, ...oneInchApi }
-        case SwapperName.Osmosis:
-          return { ...osmosisSwapper, ...osmosisApi }
-        case SwapperName.Test:
-          throw Error('not implemented')
-        default:
-          assertUnreachable(swapperName)
-      }
-    })()
+    const swapper = getSwapperBySwapperName(swapperName)
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts:77]: cleeeeannn

File: src/components/EarnDashboard/components/ProviderDetails/LpPositionsByAsset.tsx
Comment (gomesalexandre) [inline src/components/EarnDashboard/components/ProviderDetails/LpPositionsByAsset.tsx:1]: 👋🏽

File: src/components/ChainOption/ChainOption.tsx
Comment (gomesalexandre) [inline src/components/ChainOption/ChainOption.tsx:1]: Confirmed unused 👋🏽

File: src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx [hunk 4]
```diff
@@ -128,7 +126,7 @@ export const useGetTradeQuotes = () => {
           wallet,
           receiveAddress,
           sellAmountBeforeFeesCryptoPrecision: sellAmountCryptoPrecision,
-          allowMultiHop: flags.MultiHopTrades,
+          allowMultiHop: true,
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx:129]: 🎉

File: src/components/MultiHopTrade/hooks/useReceiveAddress.tsx [hunk 5]
```diff
@@ -12,6 +13,23 @@ import {
 } from 'state/slices/swappersSlice/selectors'
 import { useAppDispatch, useAppSelector } from 'state/store'
 
+const getReceiveAddress = async ({
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useReceiveAddress.tsx:16]: Confirmed she's just moved from `components/Trade/hooks/useSwapper/utils`

File: src/hooks/useLocaleFormatter/useLocaleFormatter.ts [hunk 6]
```diff
@@ -151,6 +151,7 @@ export const useLocaleFormatter = (args?: useLocaleFormatterArgs): NumberFormatt
   /**
    * Helper function to abbreviate number to truncate rather than round fractions
    * @param {number} maximumFractionDigits - truncate fraction after this number of digits. Use 0 for no fraction.
+   * @param omitDecimalTrailingZeros
```
Comment (gomesalexandre) [inline src/hooks/useLocaleFormatter/useLocaleFormatter.ts:154]: 💜

File: src/lib/swapper/api.ts [hunk 7]
```diff
@@ -250,7 +223,7 @@ export interface Swapper<T extends ChainId> {
   /**
    * Get a trade quote
    */
-  getTradeQuote(input: GetTradeQuoteInput): Promise<Result<TradeQuote<ChainId>, SwapErrorRight>>
+  getTradeQuote(input: GetTradeQuoteInput): Promise<Result<TradeQuote, SwapErrorRight>>
```
Comment (gomesalexandre) [inline src/lib/swapper/api.ts:226]: 🧠

**Review by gomesalexandre — COMMENTED** [review]
Tested locally, one smollish issue WRT Thorchain not being available as a swapping chain, otherwise LGTM:

### Quotes

Noticed no regressions, see https://github.com/shapeshift/web/pull/5018#pullrequestreview-1558549828 - tested as part of 5018 testing, branching down and testing this separately

### Preview / status

#### This diff

- Osmosis

image
image
image

- Li.Fi

image
image


- CoW

image
image

- 1inch

image
image

Not a blocker as this is already the case in prod, but we're letting the trade go through despite "Insufficient sell amount" being displayed @0xApotheosis (which conveniently allowed testing this with very small amounts!)

- ZRX

image
image

- THOR

RUNE is unavailable as a sell / buy option

#### Prod

- Osmosis

image

- Li.Fi

image
image

- CoW

image
image

- 1inch

image
image

- ZRX

image
image

- THOR

image
image

**Review by gomesalexandre — APPROVED** [review]
Fixed RUNE as a constant buy asset in 12157669ef379b9b4c4bf9ff12fad58b04a1d419:

image
image

image
image
image

image

**Pull request merged**